pkg/query/storeset.go

```diff
@@ -268,6 +271,11 @@ func NewStoreSet(
 		stores:                make(map[string]*storeRef),
 		storeStatuses:         make(map[string]*StoreStatus),
 		unhealthyStoreTimeout: unhealthyStoreTimeout,
+
+		upStatus: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "thanos_store_status",
+			Help: "Status of a given store node address.",
+		}, []string{"addr"}),
```

Comment:
Addresses have an insane high cardinality, so this should not be used for a label. e.g. a dynamic look-up that returns a list of k8s pod ips.. Instead, use the external_label set.

Reply:
```suggestion
		}, []string{"external_labels"}),
```

Reply:
Also, including the label store_type can also be useful for aggregation purposes. e.g. you have 2 unique external label sets for ruler. You only want to send a critical alert if both are down. To achieve this you can aggregate on the store_type label. Without it, it is not easily achievable.

Reply:
```suggestion
		}, []string{"external_labels", "store_type"}),
```